subsys/net/lib/lwm2m/lwm2m_obj_connmon.c

```diff
@@ -136,6 +152,10 @@ static struct lwm2m_engine_obj_inst *connmon_create(uint16_t obj_inst_id)
 			  sizeof(cellid));
 	INIT_OBJ_RES_DATA(CONNMON_SMNC, res, i, res_inst, j, &mnc, sizeof(mnc));
 	INIT_OBJ_RES_DATA(CONNMON_SMCC, res, i, res_inst, j, &mcc, sizeof(mcc));
+#if defined(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2)
+	INIT_OBJ_RES_OPTDATA(CONNMON_SIGNAL_SNR, res, i, res_inst, j);
```

Comment:
Since you created `snr` variable for this resource, you should provide a pointer here (and use `INIT_OBJ_RES_DATA` macro).

Reply:
Done